### PR TITLE
docs (microservices): add documentation for KafkaContext.heartbeat usage

### PR DESCRIPTION
**Pull request by kosh-b**
## Add documentation of KafkaContext.heartbeat usage, related to: https://github.com/nestjs/nest/pull/9954

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Docs
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: [9949](https://github.com/nestjs/nest/issues/9949) (of nestjs itself)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by Tony133 — APPROVED** [review]

File: content/microservices/kafka.md [hunk 1]
```diff
@@ -342,6 +342,26 @@ interface IncomingMessage {
 }
 ```
 
+If your endpoint involves slow processing time for each message you should consider calling `heartbeat` callback, use `getHeartbeat()` method of `KafkaContext` which will send heartbeat to the broker, as follows:
```
Comment (jmcdo29) [inline content/microservices/kafka.md:345]:
```suggestion
If your endpoint involves slow processing time for each message you should consider using the `heartbeat` callback. To do this, use the `getHeartbeat()` method of the `KafkaContext` which will send heartbeat to the broker, as follows:
```

**Review by jmcdo29 — CHANGES_REQUESTED** [review]
Slight suggestion to grammar. Otherwise looks good

**Comment by kosh-b** [comment]
Committed the suggestion:)

File: content/microservices/kafka.md [hunk 2]
```diff
@@ -342,6 +342,26 @@ interface IncomingMessage {
 }
 ```
 
+If your endpoint involves slow processing time for each message you should consider using the `heartbeat` callback. To do this, use the `getHeartbeat()` method of the `KafkaContext` which will send heartbeat to the broker, as follows:
```
Comment (kamilmysliwiec) [inline content/microservices/kafka.md:345]:
```suggestion
If your handler involves a slow processing time for each received message you should consider using the `heartbeat` callback. To retrieve the `heartbeat` function, use the `getHeartbeat()` method of the `KafkaContext`, as follows:
```

File: content/microservices/kafka.md [hunk 3]
```diff
@@ -342,6 +342,26 @@ interface IncomingMessage {
...
+@@filename()
+@MessagePattern('hero.kill.dragon')
+async killDragon(@Payload() message: KillDragonMessage, @Ctx() context: KafkaContext) {
+  const originalMessage = context.getMessage();
+  const heartbeat = context.getHeartbeat();
+  
+  // Do some slow processing:
+  await doWorkPart1();
+
+  // Send heartbeat when it's possible, in order to not exceed the sessionTimeout
```
Comment (kamilmysliwiec) [inline content/microservices/kafka.md:357]:
```suggestion
  // Send heartbeat to not exceed the sessionTimeout
```

File: content/microservices/kafka.md [hunk 4]
```diff
@@ -342,6 +342,26 @@ interface IncomingMessage {
...
+If your handler involves a slow processing time for each received message you should consider using the `heartbeat` callback. To retrieve the `heartbeat` function, use the `getHeartbeat()` method of the `KafkaContext`, as follows:
+
+```typescript
+@@filename()
+@MessagePattern('hero.kill.dragon')
+async killDragon(@Payload() message: KillDragonMessage, @Ctx() context: KafkaContext) {
+  const originalMessage = context.getMessage();
+  const heartbeat = context.getHeartbeat();
+  
+  // Do some slow processing:
```
Comment (kamilmysliwiec) [inline content/microservices/kafka.md:354]:
```suggestion
  // Do some slow processing
```

File: content/microservices/kafka.md [hunk 5]
```diff
@@ -342,6 +342,26 @@ interface IncomingMessage {
...
+  const originalMessage = context.getMessage();
+  const heartbeat = context.getHeartbeat();
+  
+  // Do some slow processing
+  await doWorkPart1();
+
+  // Send heartbeat to not exceed the sessionTimeout
+  await heartbeat();
+
+  // Do some slow processing again:
```
Comment (kamilmysliwiec) [inline content/microservices/kafka.md:360]:
```suggestion
  // Do some slow processing again
```

File: content/microservices/kafka.md [hunk 6]
```diff
@@ -342,6 +342,26 @@ interface IncomingMessage {
 }
 ```
 
+If your handler involves a slow processing time for each received message you should consider using the `heartbeat` callback. To retrieve the `heartbeat` function, use the `getHeartbeat()` method of the `KafkaContext`, as follows:
+
+```typescript
+@@filename()
+@MessagePattern('hero.kill.dragon')
+async killDragon(@Payload() message: KillDragonMessage, @Ctx() context: KafkaContext) {
+  const originalMessage = context.getMessage();
```
Comment (kamilmysliwiec) [inline content/microservices/kafka.md:351]:
```suggestion
```

**Comment by kamilmysliwiec** [comment]
lgtm

**Pull request merged**